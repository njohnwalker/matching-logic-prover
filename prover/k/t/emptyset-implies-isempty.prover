claim \implies( \and(\equals(variable("S"), emptyset), .Patterns)
             , \and(isEmpty(variable("S"), .Patterns), .Patterns)
             )
strategy search-bound(4)
