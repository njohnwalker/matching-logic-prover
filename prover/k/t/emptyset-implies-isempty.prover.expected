<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( \equals ( variable ( "S" ) , emptyset ) , .Patterns ) , \and ( \equals ( variable ( "S" ) , emptyset ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      right-unfold
    </trace>
  </goal>
</prover>
